### PR TITLE
refactor: refactor of PlayerInfoHUD adding more test coverage

### PR DESCRIPTION
**Pull request by lorux0**
## What does this PR change?

- Defined better responsabilities between view<->controller
- Controller now injects collaborators to be able to mock dependencies and isolate automatization tests
- Fixed non-working `CreateCollectibles` test (was throwing different test results depending if runned individually or not)

## How to test the changes?

1. Go to: https://play.decentraland.zone/?renderer-branch=refactor/player-info-card
2. Click on any user avatar to check its info
3. All card elements should work properly

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### CODE REVIEW

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/CatalogController/CatalogController.cs [hunk 1]
```diff
@@ -169,14 +170,18 @@ public static Promise<WearableItem> RequestWearable(string wearableId)
...
     {
         Promise<WearableItem[]> promiseResult;
 
         if (!awaitingWearablesByContextPromises.ContainsKey(OWNED_WEARABLES_CONTEXT))
         {
             promiseResult = new Promise<WearableItem[]>();
+
+            // TODO: find a better solution to avoid executing a request when running tests, mocking the running instance
+            if (EnvironmentSettings.RUNNING_TESTS) return promiseResult;
+
```
Comment (Suduck) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/CatalogController/CatalogController.cs:184]: This is necessary? I can see that this only request the wearable to the kernel and the kernel shouldn't be active in the unit test so I don't know if this help avoid the issue

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FriendsController/FriendshipStatus.cs [hunk 2]
```diff
@@ -0,0 +1,7 @@
+﻿public enum FriendshipStatus
+{
+    NONE,
```
Comment (Suduck) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FriendsController/FriendshipStatus.cs:3]: WDYT about naming this to NOT_FRIEND?

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FriendsController/IFriendsController.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+    void RequestFriendship(string friendUserId);
+    void CancelRequest(string friendUserId);
+    void AcceptFriendship(string friendUserId);
+    void RejectFriendship(string friendUserId);
+
+    event Action OnInitialized;
+    event Action<string, FriendshipAction> OnUpdateFriendship;
+    event Action<string, FriendsController.UserStatus> OnUpdateUserStatus;
+    event Action<string> OnFriendNotFound;
```
Comment (Suduck) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FriendsController/IFriendsController.cs:19]: Can we reorder this to send the actions to the top of the interface?

**Review by Suduck — APPROVED** [review]
Good Job! Just a little concerned about the RUNNING_TEST

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/CatalogController/CatalogController.cs (hunk 1 above)
Comment (lorux0) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/CatalogController/CatalogController.cs:184]: This line avoids a timeout exception produced by requesting the wearables during tests. This timeout can break any given test that is running at that specific time. I will try to find a workaround by mocking this behaviour so production code is not affected

Comment (lorux0) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/CatalogController/CatalogController.cs]: `CatalogController` now has a method `Clear` which resets all the pending promises made by any other test. This is executed on `TearDown`  at `IntegrationTestSuite_Legacy` class which is used by most of the play mode tests

**Pull request merged**